pkg/pillar/pubsub/parser.go

```diff
@@ -0,0 +1,470 @@
+// Copyright (c) 2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Define json parser functions which return a specific type
+
+package pubsub
+
+import (
+	"encoding/json"
+
+	"github.com/lf-edge/eve/pkg/pillar/types"
+)
+
+// GetParserFromType returns a function which will parse json
+// into the specific named type.
+func GetParserFromType(typeName string) types.ParseFunc {
```

Comment:
Why is this function in pubsub package? This is a layer violation :-)
This is similar to the callback I was referring to. We should pass this as an argument or as a member when creating a PubSub..

Reply:
See previous utils wrapper attempt. I don't think we want each call site of Publish* and Subscribe* to pass in the parser function.